### PR TITLE
Open Door Bolting

### PR DESCRIPTION
**Pull request by Geevies**
* Stationbounds can now once more bolt airlocks, but only if they are open. This allows them to easily permanently open airlocks for emergency purposes.

Presumably the VueUI has to be updated as well but I dunno how to do that.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Pratepresidenten** [comment]
How does this make sense? Open-bolting airlocks can be far more detrimental than close-bolting. Like exterior airlocks for example. How is this even explained? D:

**Comment by Geevies** [comment]
> 
> 
> How does this make sense? Open-bolting airlocks can be far more detrimental than close-bolting. Like exterior airlocks for example. How is this even explained? D:

Bolting open doors to let paramedics pass through. AIs are lawed against opening external airlocks, and Nanotrasen doesn't really expect many threats with ID-access on the station. That's my headcanon, anyway.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -965,12 +965,12 @@ About the new airlock wires panel:
 					to_chat(usr, SPAN_NOTICE("The door bolts have been raised."))
 		if("electrify_temporary")
 			if(issilicon(usr) && !player_is_antag(usr.mind))
-				to_chat(usr, SPAN_WARNING("Your programming prevents you from electrifying the door."))
+				to_chat(usr, SPAN_WARNING("Nanotrasen Safety Protocol #516 you from electrifying the door."))
```
Comment (Doxxmedearly) [inline code/game/machinery/doors/airlock.dm:968]:
```suggestion
				to_chat(usr, SPAN_WARNING("Nanotrasen Safety Protocol #516 prevents you from electrifying the door."))
```

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -965,12 +965,12 @@ About the new airlock wires panel:
...
 			else
 				electrify(30 * activate, 1)
 		if("electrify_permanently")
 			if(issilicon(usr) && !player_is_antag(usr.mind))
-				to_chat(usr, SPAN_WARNING("Your programming prevents you from electrifying the door."))
+				to_chat(usr, SPAN_WARNING("Nanotrasen Safety Protocol #516 you from electrifying the door."))
```
Comment (Doxxmedearly) [inline code/game/machinery/doors/airlock.dm:973]:
```suggestion
				to_chat(usr, SPAN_WARNING("Nanotrasen Safety Protocol #516 prevents you from electrifying the door."))
```

**Comment by Doxxmedearly** [comment]
idk. I get where you're coming from with the idea and it's a good reason, but can't they just disable the IDscan for this purpose?

The flavor of the Safety Protocol is rad.

**Comment by Geevies** [comment]
> 
> 
> idk. I get where you're coming from with the idea and it's a good reason, but can't they just disable the IDscan for this purpose?
> 
> The flavor of the Safety Protocol is rad.

They could, but the second lost opening the door could be crucial. It also makes pushing stuff through the doorway more difficult. I just don't see an issue with bolting doors open as much as bolting doors shut.

**Comment by Chada-1** [comment]
> > How does this make sense? Open-bolting airlocks can be far more detrimental than close-bolting. Like exterior airlocks for example. How is this even explained? D:
> 
> Bolting open doors to let paramedics pass through. AIs are lawed against opening external airlocks, and Nanotrasen doesn't really expect many threats with ID-access on the station. That's my headcanon, anyway.

I agree with this explanation, but can't the stationbound just disable the ID scan on the airlock? That's a thing and it allows anybody to open an airlock 👀

**Comment by Doxxmedearly** [comment]
> They could, but the second lost opening the door could be crucial. It also makes pushing stuff through the doorway more difficult. I just don't see an issue with bolting doors open as much as bolting doors shut.

It's not an issue of gameplay balance in this case for me. I feel the headcanon for being able to bolt them open but not closed is a little weak. That said, I don't see any other code issues and if the other devs are chill with it I think it's a good change regardless.

File: html/changelogs/geeves-open_door_bolting.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+author: Geeves
+
+delete-after: True
+
+changes: 
+  - tweak: "Stationbounds can now once more bolt airlocks, but only if they are open. This allows them to easily permanently open airlocks for emergency purposes."
```
Comment (Geevies) [inline html/changelogs/geeves-open_door_bolting.yml:6]:
```suggestion
  - tweak: "Stationbounds can now bolt airlocks open. This allows them to easily permanently open airlocks for emergency purposes."
```

File: code/game/machinery/doors/airlock.dm [hunk 4]
```diff
@@ -952,8 +952,8 @@ About the new airlock wires panel:
 			if(src.isWireCut(AIRLOCK_WIRE_DOOR_BOLTS))
 				to_chat(usr, SPAN_WARNING("The door bolt control wire is cut - Door bolts permanently dropped."))
 			else if(activate)
-				if(issilicon(usr) && !player_is_antag(usr.mind))
-					to_chat(usr, SPAN_WARNING("Your programming prevents you from lowering the door bolts."))
+				if(issilicon(usr) && density && !player_is_antag(usr.mind))
```
Comment (mikomyazaki) [inline code/game/machinery/doors/airlock.dm:955]: Is the density check just meant to exclude drones? 

Does it affect AIs that are holograms or anything? You might wanna do a more exclusive istype(usr, drone) thing.

Comment (Geevies) [reply code/game/machinery/doors/airlock.dm:955]: the density thing checks whether the door is closed or not

**Comment by Geevies** [comment]
!review

**Comment by mikomyazaki** [comment]
> Presumably the VueUI has to be updated as well but I dunno how to do that.

You'll need to figure out this, and the ui is already relatively complicated. I guess you'll need to add another var in the vueui that acts as an override for the disable when the door is open.

**Comment by Arrow768** [comment]
The UI needs to be taken care of.
That said, I am not a huge fan of adding more code to facilitate that "one second" improvement when:
- There is already a feature that can be used to do that: disabling IDscan
- Adding the ability to bolt open airlocks can enable the AI to do stupid things (such as trying to vent confirmed antagonists)

**Comment by NonQueueingMatt** [comment]
Stale.

**Pull request closed without merging**